htdocs/adherents/card.php

```diff
@@ -603,6 +603,9 @@
 					setEventMessages($object->error, $object->errors, 'errors');
 				}
 			}
+			$page = DOL_URL_ROOT.'/adherents/card.php?rowid='.$object->id;
```

Comment:
If there is an error in this action, we must not make a redirect, we must stay on the form to create. So we must force  $action = 'create' and not make the redirect

Reply:
let me look into that

Reply:
Would this be ok?

```php
if ($result < 0 || !$error) {
	$action =  '';
	$page = DOL_URL_ROOT.'/adherents/card.php?rowid='.$object->id;
	header("Location: ".$page);
	exit;
} else {
	$action = 'create';
}
```
at the end?

AND is this better?:

```php
header("Location: ".$_SERVER["PHP_SELF"].'?rowid='.$object->id);
```

Reply:
If ($result < 0
it means there was an error too, isn't it ? In such a case, we should not make the redirect. redirect should be done on success only.

Reply:
May be